lib/source_file.cpp

```diff
@@ -242,13 +265,29 @@ cstring SourceInfo::toSourceFragment() const {
     return InputSources::instance->getSourceFragment(*this);
 }
 
+cstring SourceInfo::toSourceFragment2() const {
+    return InputSources::instance->getSourceFragment2(*this);
+}
+
 cstring SourceInfo::toPositionString() const {
     if (!this->isValid())
         return "";
     SourceFileLine position = InputSources::instance->getSourceLine(this->start.getLineNumber());
     return position.toString();
 }
 
+cstring SourceInfo::toSourcePositionData(unsigned *outLineNumber,
+                                         unsigned *outColumnNumber) const {
+    SourceFileLine position = InputSources::instance->getSourceLine(this->start.getLineNumber());
```

Comment:
Doh!  OK, thanks for that.  I will now try to hit the big scary Github "Update branch" button and see if that causes the state of this PR to become unreadable.  If so, I will create a new separate PR that is clean but won't have all the nice review comments on it.